zio-http/src/main/scala/zio/http/api/openapi/JsonRenderer.scala

```diff
@@ -0,0 +1,57 @@
+package zio.http.api.openapi
+
+import zio.http.api.Doc
+import zio.http.model.Status
+
+import java.net.URI
+import scala.collection.immutable.Iterable
+
+private[openapi] object JsonRenderer {
+
+  def renderFields(fieldsIt: (String, Any)*): String = {
+    if (fieldsIt.map(_._1).toSet.size != fieldsIt.size) {
+      throw new IllegalArgumentException("Duplicate field names")
+    } else {
+      val fields = fieldsIt
+        .filterNot(_._2 == None) // We don't render empty fields
+        .map { case (name, value) => s""""$name":${renderValue(value)}""" }
+      s"{${fields.mkString(",")}}"
+    }
+  }
+
+  private def renderValue(value: Any): String = value match {
```

Comment:
Can we make this more type safe, i.e. define `trait Renderable[A]` or something?

Reply:
I'll do that.

Reply:
@jdegoes I just realized, I am not sure I can. Please take a look at the `Link` class. It has fields of type any.

Reply:
The spec says actually `Any`. But maybe we have to be more restrictive? Also, what/how would you read this, if you want to impl. #1520

Reply:
https://swagger.io/specification/#Link_Object

Reply:
So we are missing a `LinkObject` which is a `Map[String, LinkObject | Reference]`